packages/hash/frontend/src/components/BlockContextMenu/BlockLoaderInput.tsx

```diff
@@ -27,7 +23,7 @@ export const BlockLoaderInput: React.VFC = () => {
       .fetchAndDefineBlock(blockUrl)
       .then((blockMeta) => {
         setError(null);
-        setBlocksMeta(setEntry(blocksMeta, [blockUrl, blockMeta]));
+        setBlocksMeta({ ...blocksMeta, [blockUrl]: blockMeta });
```

Comment:
Potentially minor issue, but this will use the `blocksMeta` from when the request began – not from the time when you're setting it. This means you're at risk of reverting changes if multiple requests to load new blocks (or other changes to `blocksMeta`) are happening at once, with a race condition. 

You can fix this by tracking a current version using a ref, or by using the functional `setState` (which is passed the latest value as an argument`), though you'd have to update the provider typings for that.

Reply:
Done.